pkg/framework/resource/resource.go

```diff
@@ -0,0 +1,59 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resource
+
+import (
+	"k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	openapi "k8s.io/kube-openapi/pkg/util/proto"
+)
+
+// Resource is an API Resource.
+type Resource struct {
+	Resource        v1.APIResource
```

Comment:
I think we'd rather make these private by default, and change them to public as needed.

Reply:
You sure? If I were to do this it'd break my filter examples (they depend on r.Resource.Name).

I tried it anyway and the IDE gave me:

> Rename failed: /home/ellen/go/src/k8s.io/kubectl/pkg/framework/resource/resource.go:26:6: renaming this type "Resource" to "resource" would make it unexported /home/ellen/go/src/k8s.io/kubectl/pkg/framework/resource/poc/main.go:38:38: breaking references from packages such as "k8s.io/kubectl/pkg/framework/resource/poc" [...]

I did it by hand then and indeed it broke everything.

> undefined: "k8s.io/kubectl/pkg/framework/resource".Resource

There's probably a way to do the filtering differently, but do we really have to?

Reply:
Sorry if I was unclear. By "these", I meant "all these fields". If this one in particular is needed, let's keep it public, what about the others?